### ISSUE TITLE
Encryption standards for V.systems

### ISSUE THREAD

**Issue by Primerz** [virtualeconomy/js-v-sdk#59]
As explained by the Vsys team (here https://github.com/virtualeconomy/js-v-sdk/wiki/How-to-Encrypt-and-Decrypt-a-message) it is possible to encrypt a message in a transaction using js-nacl, the key-pair from the vsystems account are compatible with it, meanwhile, for the McEliece algorithm it is not specified how could we transfers the public key to someone, therefore, it is potentially, useless.

In some chain, it is possible to have a transaction object property called "is_encrypted" which can correlate to "nacl" for example, very good for a standardized way to use it.

Also, a question that I have is the size limit of the message embed in a transaction.

**Comment by Primerz** [comment]
UPDATE: NaCl has been implemented on https://wallet.crypto.red/, it works and the string result is somehow sometimes smaller than 160 characters.

Since there is no property is_message_encrypted or such, the web app mentioned above offer a "tool" not a module which detect and decrypt the message, why not creating instruction like "E_NACL {{encrypted_text}}", it takes 7 chars... and "E_MCLE ..." too.

**Comment by Primerz** [comment]
```
function mc_eliece_encrypt(message, public_key, callback_function) {

    const message_uint8array = new TextEncoder().encode(message);
    const public_key_uint8array = new TextEncoder().encode(public_key);

    mceliece.encrypt(message_uint8array, public_key_uint8array).then(function(response_uint8array){

        callback_function(null, base58.encode(response_uint8array));

    }).catch(function(error){

        callback_function(error, "");
    });
}

function mc_eliece_decrypt(encrypted_message, private_key, callback_function) {

    const encrypted_message_uint8array = new TextEncoder().encode(encrypted_message);
    const private_key_uint8array = new TextEncoder().encode(private_key);

    mceliece.decrypt(encrypted_message_uint8array, private_key_uint8array).then(function(response_uint8array){

        callback_function(null, base58.encode(response_uint8array));

    }).catch(function(error){

        callback_function(error, "");
    });
}
```

I am not sure the keypair of Vsys is compatible with McEliece...

**Comment by faddat** [comment]
@Primerz can you please let me know the goal?

passing encrypted info between users or something else?

**Comment by vipertechofficial** [comment]
@faddat Yes to encrypt the transaction's message, McEliece doesn't seem to fit (the VSYS key-pair can't achieve to work with it), more than that, it isn't recommended to use McEliece to encrypt messages, rather it should be used to encrypt a symmetrical key.

So we probably should remove McEliece from the recommendation, meanwhile, NaCl works great, cool enough, the vsystems blockchain could provide a format such as "NACL:{encrypted_message}" in order for all application to recognize encrypted memo.

**Comment by Primerz** [comment]
![Screenshot from 2021-06-30 22-55-57](https://user-images.githubusercontent.com/19823160/124030813-d5f94f00-d9f6-11eb-9133-491f31d593ad.png)
![Screenshot from 2021-06-30 22-56-46](https://user-images.githubusercontent.com/19823160/124030823-d8f43f80-d9f6-11eb-9464-ca89a0790ac1.png)
App screenshot: https://github.com/crypto-red/crypto-red.github.io

I think an **auto decrypt mechanism** along with a way to encrypt message is recommendable, also can't deal with McEliece by the way nevertheless NaCl works fine. :)
